File: backend/models/bookmark.py
from models.database import db
from datetime import datetime

class Bookmark(db.Model):
    __tablename__ = 'bookmarks'
    
    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
    document_id = db.Column(db.Integer, db.ForeignKey('documents.id'), nullable=False)
    notes = db.Column(db.Text)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    def to_dict(self):
        return {
            'id': self.id,
            'document_id': self.document_id,
            'notes': self.notes,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'document': self.document.to_dict() if self.document else None
        }

File: src/JuriIQ.Core/Entities/Bookmark.cs
namespace JuriIQ.Core.Entities;

public class Bookmark
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public int DocumentId { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; }

    // Navigation properties
    public User User { get; set; } = null!;
    public Document Document { get; set; } = null!;
}
